File: 04/main.hs
main = do
    putStrLn "Enter a noun:"
    noun <- getLine
    putStrLn "Enter a verb:"
    verb <- getLine
    putStrLn "Enter an adjective:"
    adj <- getLine
    putStrLn "Enter an adverb:"
    adv <- getLine
    putStrLn ("Do you " ++ verb ++ " your " ++ adj ++ " " ++ noun ++ " " ++ adv ++ "? That's hilarious!")

File: 04/main.clj
(println "Enter a noun:")
(let [noun (read-line)]
  (println "Enter a verb:")
  (let [verb (read-line)]
    (println "Enter an adjective:")
    (let [adj (read-line)]
      (println "Enter an adverb:")
      (let [adv (read-line)]
        (println (str "Do you " verb " your " adj " " noun " " adv "? That's hilarious!"))))))
